spring-boot-project/spring-boot/src/main/java/org/springframework/boot/context/properties/source/SystemEnvironmentPropertyMapper.java

```diff
@@ -116,13 +116,24 @@ private boolean isLegacyAncestorOf(ConfigurationPropertyName name, Configuration
 		if (!hasDashedEntries(name)) {
 			return false;
 		}
+		StringBuilder legacyCompatibleName = buildLegacyCompatibleName(name);
+		try {
+			return ConfigurationPropertyName.of(legacyCompatibleName).isAncestorOf(candidate);
+		}
+		catch (Exception ex) {
```

Comment:
Can you share what that extra catch block is supposed to cover?

Reply:
The removal of `isValid` here might lead to exceptions if the name can't be constructed.

Reply:
I'd argue that `legacyCompatibleName` is a bit misleading then. It makes me wonder looking at the code why a catch block is necessary. Thoughts?

Reply:
I didn't change anything in that regards. I think the name was chosen to reflect the "legacy" format and `isValid()` was added to be sure that no incompatible output breaks the code here. This PR just sticks to that behaviour of not breaking anything with a potentially invalid name.
Having that said, I too had a hard time seeing how an invalid name can be produced from a valid name, but didn't want to change the intention of the code.

Reply:
Alright, thanks for the feedback. Flagging for team attention to see what the rest of the team thinks.